apps/drec-api/src/utils/group-by-properties.ts

```diff
@@ -0,0 +1,48 @@
+export const arrayFromObject = <T>(obj: T): T[Extract<keyof T, string>][] => {
+  const arr = [];
+  for (const i in obj) {
+    arr.push(obj[i]);
+  }
+  return arr;
+};
+
+export const groupByProperties = <T, U, K>(
+  list: U[],
+  callback: (listItem: U) => K[],
+): T[Extract<keyof T, string>][] => {
+  const groups: any = {} as T;
+  for (let i = 0; i < list.length; i++) {
+    const group: string = JSON.stringify(callback(list[i]));
+    if (group in groups) {
+      groups[group].push(list[i]);
+    } else {
+      groups[group] = [list[i]];
+    }
+  }
+  return arrayFromObject(groups);
+};
+
+export const groupByProps = <T, U, K>(
+  array: T[],
+  callback: (listItem: T) => U[],
+): K[] => {
+  const groups: any = {};
+  array.forEach((item): void => {
+    const group: string = JSON.stringify(callback(item));
+    groups[group] = groups[group] || [];
+    groups[group].push(item);
+  });
+  return Object.keys(groups).map((group) => {
+    return groups[group];
+  });
+};
+
+export const hasOwnProperty = <
```

Comment:
For type safe `hasOwnProperty` you can use `in` in some cases (for example `if ('someKey' in obj)`).
It works for union types to determine particular type for example. Just saying, maybe that's your case.

Is this function even used anywhere?

Reply:
Now it`s not used anymore. Thought I leave it as an util function.

Reply:
Won't push that, but what's the point in leaving unused code?